src/pages/messages/TasksPage.tsx

```diff
@@ -188,28 +193,33 @@ const TasksPage = () => {
 
   const [nviYearFilter, setNviYearFilter] = useState(nviYearFilterValues[1]);
 
-  const nviSearchQuery = queryParam ? `&query=${queryParam}` : '';
-
-  const nviAssigneeQuery = showOnlyMyTasks && nvaUsername ? `&assignee=${nvaUsername}` : '';
-
-  const nviAggregationQuery = `year=${nviYearFilter}&affiliations=${organizationScope.join(
-    ','
-  )}${excludeSubunitsQuery}${nviAssigneeQuery}${nviSearchQuery}`;
-  const nviListQuery = `${nviAggregationQuery}&filter=${nviStatusFilter}`;
-
-  const nviAggregationsQuery = useQuery({
+  const nviAggregationsQuery = useFetchNviCandidates({
     enabled: isOnNviCandidatesPage || isOnNviStatusPage,
-    queryKey: ['nviCandidates', 1, 0, nviAggregationQuery],
-    queryFn: () => fetchNviCandidates(1, 0, nviAggregationQuery),
-    meta: { errorMessage: t('feedback.error.get_nvi_candidates') },
+    params: {
+      size: 1,
+      aggregation: 'all',
+      year: nviYearFilter,
+      affiliations: organizationScope,
+      excludeSubUnits: excludeSubunits,
+      assignee: showOnlyMyTasks && nvaUsername ? nvaUsername : null,
+      query: queryParam,
+    },
   });
```

Comment:
Her er et eksempel på det jeg snakket om i kommentaren jeg skrev i useFetchNviCandidates.ts: Vi bruker nå en hook, men det er like mange linjer kode, og fortsatt veldig mye logikk som beskriver hvordan dataene hentes. Jeg syns heller vi bør lage en mer spesifikk hook som innkapsler dette params-objektet, enten som en erstatning for den generiske `useFetchNviCandidates`, eller som et nivå utenfor som bruker `useFetchNviCandidates` og primært er der for å legge på konfigurasjonen. Fordelen med dette er at det spesifikke navnet vil også forklare for en utvikler som leser koden hva som er forskjellen på denne kallet og kallet under, som begge bruker `useFetchNviCandidates` (altså ala `useFetchOrganizationCandidates` eller `useFetchAllCandidates` eller liknende, avhengig av hva som er spesielt med akkurat dette søket).

**Målet bør være at søket gjøres på kun en linje.** Slik at komponenten primært handler om andre ting enn API-kall.

Reply:
Kan prøve med params inni hooken da! Jeg er skeptisk til en hook per bruk i dette tilfellet, da de alltid vil være veldig like og avhenge av hva brukeren søker på. Det som skal overstyres blir (i noen/ett(?)) tilfeller paginering og aggregeringer, men selv da er formålet fortsatt det samme som ellers, nemlig å søke opp NVI-kandidater. Men virker som vi møtes i middle ground med i flytte params også til hooken, så tester det!
Blir noe ganske likt det andre eksemplet i den forrige kommentaren min her da